test/missing.jl

```diff
@@ -188,16 +188,23 @@ Base.one(::Type{Unit}) = 1
 end
 
 @testset "rounding functions" begin
-    rounding_functions = [ceil, floor, round, trunc]
-
     # All rounding functions return missing when evaluating missing as first argument
+    @test ismissing(round(missing, RoundToZero))
+    @test ismissing(round(Union{Int, Missing}, missing, RoundToZero))
+    @test ismissing.(round.([1.0, missing], RoundToZero)) == [false, true]
+    @test ismissing.(round.(Union{Int, Missing}, [1.0, missing], RoundToZero)) == [false, true]
+
+    rounding_functions = [ceil, floor, round, trunc]
     for f in rounding_functions
         @test ismissing(f(missing))
-        @test ismissing(f(missing, 1))
-        @test ismissing(f(missing, 1, 1))
+        @test ismissing(f(missing, sigdigits=1, digits=1, base=1))
         @test ismissing(f(Union{Int, Missing}, missing))
         @test f(Union{Int, Missing}, 1.0) === 1
```

Comment:
This test should also have a variant with keyword arguments, passing it a value which allows testing that the result has the intended number of digits. That way we're sure the arguments are actually passed.

Reply:
The versions which accept keyword arguments never call round(). They only need to accept them so that elementwise works on mixed arrays.

Reply:
No, that's not how it works. Element-wise operation just calls the method corresponding to a given element. It doesn't depend on the fact that they are stored in an array at all.

And something like `round(Union{Int, Missing}, 1.11111, digits=2)` *has* to pass keyword arguments to `round` down the stream.

Reply:
> No, that's not how it works. Element-wise operation just calls the method corresponding to a given element. It doesn't depend on the fact that they are stored in an array at all.

I realize that. That's why `round(::Missing; sigdigits::Integer=1, ...)` doesn't need to call `round()` and just returns `missing`.

> And something like `round(Union{Int, Missing}, 1.11111, digits=2)` _has_ to pass keyword arguments to `round` down the stream.

That's not a valid call to `round()`. The type argument doesn't mix with the keyword arguments (although perhaps it should, especially in the case of `sigdigits`...).

Reply:
Ah, sorry, there are so many signatures.

Reply:
Tell me about it...